### ISSUE TITLE
alternator: improve error message in case of forbid_rmw forbidden write

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#6421]
One of the _write isolation policies_ we support (see `docs/alternator/alternator.md`) is `forbid_rmw`, which forbids writes which require a prior read (e.g., conditional updates).
In `alternator/executor.cc` we have:
```c++
    if (needs_read_before_write) {
        if (_write_isolation == write_isolation::FORBID_RMW) {
            throw api_error("ValidationException", "Read-modify-write operations not supported");
        }
```
The message is correct, but could have been more helpful to non-expert users. Instead of "not supported" we should say that it is not allowed because of the `forbid_rmw` setting, and that it can be allowed by setting the `system:write_isolation` tag on the table to `always`.

**Comment by nyh** [comment]
Backported to next-4.1 (6d1301d93ce814b61d7f5c73f895585d9dc5ab5c), would be useful together with the fix to #6452 which would also be backported to 4.1
Not backporting to 4.0 where the default will remain always_lwt (originally we wanted to do this for 4.0.1, but since 4.1 is coming out quickly, better do it in 4.1 instead).

**Comment by nyh** [comment]
Bot erroneously re-added "backport candidate" label after it was already removed. Re-removing it.
